### ISSUE TITLE
Reset on Drum Sequencer changes something..... Not Quite there yet

### ISSUE THREAD

**Issue by Petervos2018** [JerrySievert/DrumKit#9]
Hi Jerry,
Open this patch change the first row or any row press reset and watch what happens.
Try it a few times, it resets the edit or something....
[Jerry S Drums - 28102019-2B-Jerry.zip](https://github.com/JerrySievert/DrumKit/files/3791638/Jerry.S.Drums.-.28102019-2B-Jerry.zip)

Greetings Peter.

**Comment by JerrySievert** [comment]
hrm. let me rethink exactly how I want it to behave (I'll consult the manual I wrote, see if I spelled it out well enough).

obviously neither this nor the other behavior is correct.

I *think* the intended behavior is that hitting reset when `cycle` is active would save the existing patch, then reset to the patch in position 0.  I'll double check my notes, and if that's the case I can make a change in one line.

I've written too many modules since I released this :)

**Comment by Petervos2018** [comment]
I thought "reset" would mean go to the beginning of the pattern or seq if in seq mode ?
Not "reset what you have done". I think that is called "undo" ;)

Greeting Peter

**Comment by JerrySievert** [comment]
reset does go to the beginning of the pattern, yes, but what I need to remember is my intent of the following scenario:

1. cycle active
2. not on primary pattern
3. edits made to current pattern

if memory serves, the proper sequence of events is:

1. save edits of the current pattern
2. switch the underlying pattern to sequence position 1's pattern

previously, it was using an internal "shortcut" to switch the underlying pattern, which was causing it to overwrite the pattern.  I switched it to use the same mechanism that a pattern change uses (not the shortcut), but apparently missed the part that saves the edit.  like I said, been a little too long since I looked at that code :)
